recipes/bilby_tgr/meta.yaml

```
@@ -0,0 +1,48 @@
+{% set name = "bilby_tgr" %}
+{% set version = "0.1a1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/bilby_tgr-{{ version }}.tar.gz
+  sha256: 7074eeff19da652449a2e2539d932506ffad3dd8ccde5cd9c5eebb1fe15b4613
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - setuptools >=42
+    - setuptools-scm >=3.4.3
+    - wheel
+    - pip
+  run:
+    - python >=3.8
+    - bilby
+    - lalsuite
```

It _may_ be more efficient to replace the `lalsuite` requirement with the list of specific subcomponent you need, to allow users to build a much smaller environment.

```suggestion
    - python-lal
    - python-lalsimulation
```

However, this comes at a (small) cost that you will need to excise `lalsuite` from the `requirements.txt` as a [patch](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#patches) to the conda recipe.

```patch
diff --git a/requirements.txt b/requirements.txt
index f32e630..fb6da21 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,2 @@
 bilby
-lalsuite
 numpy>=1.17
```

---

Thanks @duncanmmacleod . Should we also excise `lalsuite` from the project dependencies list in `pyproject.toml` ? Just to make sure I understand the patching process, this change will not have to eventually propagate to the source repo but is applied on top of it for the conda build, correct? (otherwise it breaks the pipeline tests)

---

@magathos, if you have project metadata in `pyproject.toml` you should _remove_ the `setup.py` file completely. Until that time, yes, you need to patch `pyproject.toml` as well.

As for the patch, yes, this is applied on top of the conda build and should not be applied to the source repo.